addons/mail/static/src/models/composer_view/composer_view.js

```diff
@@ -132,23 +131,23 @@ registerModel({
                 );
             }
             const recordReplacement = this.activeSuggestedRecord.getMentionText();
-            const updateData = {
-                isLastStateChangeProgrammatic: true,
-                textInputContent: textLeft + recordReplacement + ' ' + textRight,
-                textInputCursorEnd: textLeft.length + recordReplacement.length + 1,
-                textInputCursorStart: textLeft.length + recordReplacement.length + 1,
-            };
             // Specific cases for channel and partner mentions: the message with
             // the mention will appear in the target channel, or be notified to
             // the target partner.
             switch (this.activeSuggestedRecord.constructor.name) {
                 case 'Thread':
-                    Object.assign(updateData, { mentionedChannels: link(this.activeSuggestedRecord) });
+                    this.composer.messageComposition.update({ mentionedChannels: link(this.activeSuggestedRecord) });
                     break;
                 case 'Partner':
-                    Object.assign(updateData, { mentionedPartners: link(this.activeSuggestedRecord) });
+                    this.composer.messageComposition.update({ mentionedPartners: link(this.activeSuggestedRecord) });
                     break;
             }
+            const updateData = {
+                isLastStateChangeProgrammatic: true,
+                textInputContent: textLeft + recordReplacement + ' ' + textRight,
+                textInputCursorEnd: textLeft.length + recordReplacement.length + 1,
+                textInputCursorStart: textLeft.length + recordReplacement.length + 1,
+            };
             this.composer.update(updateData);
```

Comment:
Could just `composer.update({ ... })`. The variable `updateData` is no more necessary.